packages/dialtone-vue2/components/toast/toast.vue

```diff
@@ -1,60 +1,81 @@
   >
-    <div class="d-toast__dialog">
-      <dt-notice-icon
-        v-if="!hideIcon"
-        :kind="kind"
-        v-on="$listeners"
-      >
-        <!-- @slot Slot for custom icon -->
-        <slot name="icon" />
-      </dt-notice-icon>
-      <dt-notice-content
-        :title-id="titleId"
-        :content-id="contentId"
-        :title="title"
-        :role="role"
-        v-on="$listeners"
-      >
-        <template #titleOverride>
-          <!-- @slot Allows you to override the title, only use this if you need to override
+    <!-- @slot Slot for custom icon -->
+    <template #icon>
+      <slot name="icon" />
+    </template>
+    <template #titleOverride>
+      <!-- @slot Allows you to override the title, only use this if you need to override
           with something other than text. Otherwise use the "title" prop. -->
-          <slot name="titleOverride" />
-        </template>
-        <!-- @slot the main textual content of the toast -->
-        <slot>
-          {{ message }}
-        </slot>
-      </dt-notice-content>
-      <dt-notice-action
-        :hide-action="hideAction"
-        :hide-close="hideClose"
-        :close-button-props="closeButtonProps"
-        :visually-hidden-close="visuallyHiddenClose"
-        :visually-hidden-close-label="visuallyHiddenCloseLabel"
-        v-on="noticeActionListeners"
-      >
-        <!-- @slot Enter a possible action for the user to take, such as a link to another page -->
-        <slot name="action" />
-      </dt-notice-action>
-    </div>
-  </div>
+      <slot name="titleOverride" />
+    </template>
+    <!-- @slot the main textual content of the toast -->
+    <slot>
+      {{ message }}
+    </slot>
+    <!-- @slot Enter a possible action for the user to take, such as a link to another page -->
+    <template #action>
+      <slot name="action" />
+    </template>
+  </toast-layout-alternate>
+  <toast-layout-default
+    v-else
+    :is-shown="isShown"
+    :title-id="titleId"
+    :content-id="contentId"
+    :title="title"
+    :message="message"
+    :role="role"
+    :kind="kind"
+    :important="important"
+    :close-button-props="closeButtonProps"
+    :hide-close="hideClose"
+    :hide-icon="hideIcon"
+    :hide-action="hideAction"
+    v-on="$listeners"
+    @close="handleClose"
+  >
+    <!-- @slot Slot for custom icon -->
+    <template #icon>
+      <slot name="icon" />
+    </template>
+    <template #titleOverride>
+      <!-- @slot Allows you to override the title, only use this if you need to override
+          with something other than text. Otherwise use the "title" prop. -->
+      <slot name="titleOverride" />
+    </template>
+    <!-- @slot the main textual content of the toast -->
+    <slot>
+      {{ message }}
+    </slot>
+    <!-- @slot Enter a possible action for the user to take, such as a link to another page -->
+    <template #action>
+      <slot name="action" />
+    </template>
+  </toast-layout-default>
```

Comment:
Both components have exactly the same props and slots, isn't better to use `<component :is="toastKind">` ? to avoid duplication.

Reply:
Hmm yes that's a good idea